app/policies/api/submission_file_policy.rb

```diff
@@ -4,5 +4,9 @@ class SubmissionFilePolicy < MainApiPolicy
     def index?
       real_user.autotest_user? || check?(:manage?)
     end
+
+    def submit_file?
+      role&.student? || false
```

Comment:
I'm not sure what the `|| false` is for? Does returning `nil` not cause the policy to reject the request?